scripts/needs_testrun.py

```diff
@@ -151,6 +151,11 @@ def for_each_testrun_needed(suites: t.List[str], action: t.Callable[[str], None]
             action(suite)
 
 
+def pr_matches_patterns(patterns: t.Set[str]) -> bool:
```

Comment:
Just a question - how do we determine which prechecks are necessary?

Reply:
See https://github.com/DataDog/dd-trace-py/pull/6516#discussion_r1276883261